frontend/app-development/features/dataModelling/hooks/useValidateSchemaName.ts

```diff
@@ -0,0 +1,142 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppMetadataQuery } from 'app-shared/hooks/queries';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import type { DataModelMetadata } from 'app-shared/types/DataModelMetadata';
+import type { ApplicationMetadata, DataTypeElement } from 'app-shared/types/ApplicationMetadata';
+import { extractModelNamesFromMetadataList } from '../../../utils/metadataUtils';
+
+export const useValidateSchemaName = (dataModels: DataModelMetadata[]) => {
+  const [nameError, setNameError] = useState('');
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: appMetadata } = useAppMetadataQuery(org, app);
+  const { t } = useTranslation();
+
+  const validateName = (name: string): void => {
+    if (!name) {
+      setNameError(t('validation_errors.required'));
+      return;
+    }
+    if (!name.match(SCHEMA_NAME_REGEX)) {
+      setNameError(t('schema_editor.error_invalid_datamodel_name'));
+      return;
+    }
+    if (name.length > SCHEMA_NAME_MAX_LENGTH) {
+      setNameError(t('validation_errors.maxLength', { 0: SCHEMA_NAME_MAX_LENGTH }));
```

Comment:
can we use a more descriptive variable name than `0` here?
f.ex. `maxCountCharacters` or something

Reply:
All these texts use the same variable name. For consistency's sake, can we use `number`?
![bilde](https://github.com/user-attachments/assets/450b740b-c475-4720-8827-af9cde75929e)

Reply:
Hm, yes, I see that now myself. Nice that you took an extra look!

Seems like the keys are used in `packages/shared/src/components/FormField/FormField.tsx`, but that code was way too complicated to analyze and fix in a "håndvending" 😅 Looks like the errorCode is (sometimes) fetched from an external library, ajv, that analyses the json schema of the layout maybe? But from what I can see, the variable is never actually added to the i18next tho. 

But I agree, lets just keep are hands off this and adapt only the variable name in your code?